test/sdn_migration/sdn_migration_test.go

```diff
@@ -0,0 +1,639 @@
+		}
+	}
+
+	return criticalAlertCount, warningAlertCount, nil
+}
+
+// getEnvVar returns the env variable value and if unset returns default provided
+func getEnvVar(key, value string) string {
+	result, exist := os.LookupEnv(key)
+	if exist {
+		return result
+	}
+	return value
+}
+
+// osdClusterReadyHealthCheck verifies the osd-cluster-ready health check job is passing
+func osdClusterReadyHealthCheck(ctx context.Context, clusterClient *openshiftclient.Client, action, reportDir string) error {
+	var (
+		err error
+		job batchv1.Job
+	)
+
+	if err = clusterClient.Get(ctx, osdClusterReadyJobName, "openshift-monitoring", &job); err != nil {
+		return fmt.Errorf("failed to get existing %s job %v", osdClusterReadyJobName, err)
+	}
+
+	newJob := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: osdClusterReadyJobName,
+			Annotations:  job.Annotations,
+			Namespace:    job.Namespace,
+		},
+		Spec: job.Spec,
+	}
+
+	newJob.Spec.Selector.MatchLabels = map[string]string{}
+	newJob.Spec.Template.ObjectMeta.Name = newJob.GetGenerateName()
+	newJob.Spec.Template.ObjectMeta.Labels = map[string]string{}
+	newJob.Spec.Template.Spec.Containers[0].Name = newJob.GetGenerateName()
+
+	if err = clusterClient.Create(ctx, newJob); err != nil {
+		return fmt.Errorf("failed to create %s job: %v", newJob.GetName(), err)
+	}
+
+	defer func() {
+		_ = clusterClient.Delete(ctx, newJob)
+	}()
+
+	return clusterClient.OSDClusterHealthy(ctx, newJob.GetName(), reportDir, osdClusterReadyJobTimeout)
+}
+
+// getKubernetesDynamicClient returns the kubernetes dynamic client
+func getKubernetesDynamicClient(client *openshiftclient.Client) (*dynamic.DynamicClient, error) {
+	dynamicClient, err := dynamic.NewForConfig(client.GetConfig())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kubernetes dynamic client: %w", err)
+	}
+	return dynamicClient, nil
+}
+
+type upgradeError struct {
+	err error
+}
+
+// Error returns the formatted error message when upgradeError is invoked
+func (e *upgradeError) Error() string {
+	return fmt.Sprintf("osd upgrade failed: %v", e.err)
+}
+
+func getOpenshiftConfig(ctx context.Context, dynamicClient *dynamic.DynamicClient, resource string, name string) (*unstructured.Unstructured, error) {
+	gvr := schema.GroupVersionResource{
+		Group:    "config.openshift.io",
+		Version:  "v1",
+		Resource: resource,
+	}
+	return dynamicClient.Resource(gvr).Get(ctx, name, metav1.GetOptions{})
+}
+
+// checkMigrationStatus probes the status of the SDN-to-OVN migration
+func checkMigrationStatus(ctx context.Context, client *openshiftclient.Client) error {
+	var (
+		dynamicClient *dynamic.DynamicClient
+		err           error
+	)
+
+	if dynamicClient, err = getKubernetesDynamicClient(client); err != nil {
+		return &upgradeError{err: err}
+	}
+	for i := 1; i <= upgradeMaxAttempts; i++ {
+		// Get the current network configuration
+		networkConfig, err := getOpenshiftConfig(ctx, dynamicClient, "networks", "cluster")
```

Comment:
You should be able to do a this instead of the dynamic client

```
    var network configv1.Network
    _ = oc.Get(context.Background(), "", "cluster", &network)
```